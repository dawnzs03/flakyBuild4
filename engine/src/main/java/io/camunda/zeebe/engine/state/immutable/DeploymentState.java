/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
 * one or more contributor license agreements. See the NOTICE file distributed
 * with this work for additional information regarding copyright ownership.
 * Licensed under the Zeebe Community License 1.1. You may not use this file
 * except in compliance with the Zeebe Community License 1.1.
 */
package io.camunda.zeebe.engine.state.immutable;

import io.camunda.zeebe.protocol.impl.record.value.deployment.DeploymentRecord;
import org.agrona.DirectBuffer;

public interface DeploymentState {

  /**
   * Returns whether there are any deployment distributions pending for a deployment.
   *
   * @param deploymentKey the key of the deployment that may have a pending distribution
   * @return {@code true} if a pending deployment for the deployment key exists, otherwise {@code
   *     false}.
   */
  boolean hasPendingDeploymentDistribution(long deploymentKey);

  /**
   * Returns whether a specific deployment distribution for a specific partition is pending.
   *
   * @param deploymentKey the key of the deployment that may have a pending distribution
   * @param partitionId the id of the partition to which the distribution might be pending
   * @return {@code true} if the specific pending deployment exists, otherwise {@code false}.
   */
  boolean hasPendingDeploymentDistribution(long deploymentKey, int partitionId);

  DeploymentRecord getStoredDeploymentRecord(long deploymentKey);

  void foreachPendingDeploymentDistribution(PendingDeploymentVisitor pendingDeploymentVisitor);

  @FunctionalInterface
  interface PendingDeploymentVisitor {
    void visit(final long deploymentKey, final int partitionId, final DirectBuffer directBuffer);
  }
}
