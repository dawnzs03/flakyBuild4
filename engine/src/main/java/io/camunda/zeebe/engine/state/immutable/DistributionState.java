/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
 * one or more contributor license agreements. See the NOTICE file distributed
 * with this work for additional information regarding copyright ownership.
 * Licensed under the Zeebe Community License 1.1. You may not use this file
 * except in compliance with the Zeebe Community License 1.1.
 */
package io.camunda.zeebe.engine.state.immutable;

import io.camunda.zeebe.protocol.impl.record.value.distribution.CommandDistributionRecord;

public interface DistributionState {

  /**
   * Returns whether there are any distributions pending for a given key.
   *
   * @param distributionKey the key of the distribution
   * @return true if there are pending distributions for the given key, otherwise false
   */
  boolean hasPendingDistribution(long distributionKey);

  /**
   * Returns whether a specific distribution for a specific partition is pending.
   *
   * @param distributionKey the key of the distribution that may be pending
   * @param partition the id of the partition for which the distribution might be pending
   * @return {@code true} if the specific pending distribution exists, otherwise {@code false}.
   */
  boolean hasPendingDistribution(long distributionKey, int partition);

  /**
   * Returns the {@link CommandDistributionRecord} for the given distribution key. This method takes
   * a partition id. This is only used to set the partition property in the {@link
   * CommandDistributionRecord}. Doing so allows us to return a whole record, without the need to
   * remember setting the partition everytime this method is called.
   *
   * @param distributionKey the key of the distribution
   * @param partition the partition to distribute to
   * @return an new instance of the {@link CommandDistributionRecord}
   */
  CommandDistributionRecord getCommandDistributionRecord(long distributionKey, int partition);

  /**
   * Visits each persisted pending distribution, providing both the key of that distribution and the
   * {@link CommandDistributionRecord}.
   *
   * <p>Note that a new instance of the record is provided for each visit, so the visitor does not
   * have to make a copy when long term access is needed.
   *
   * @param visitor Each pending distribution is visited by this visitor
   */
  void foreachPendingDistribution(PendingDistributionVisitor visitor);

  /** This visitor can visit pending distributions of {@link CommandDistributionRecord}. */
  @FunctionalInterface
  interface PendingDistributionVisitor {

    /**
     * Visits a pending distribution.
     *
     * @param distributionKey The key of the pending distribution
     * @param pendingDistribution The pending distribution itself as command distribution record
     */
    void visit(final long distributionKey, final CommandDistributionRecord pendingDistribution);
  }
}
