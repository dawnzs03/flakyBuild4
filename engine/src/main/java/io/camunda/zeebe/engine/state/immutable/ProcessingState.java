/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
 * one or more contributor license agreements. See the NOTICE file distributed
 * with this work for additional information regarding copyright ownership.
 * Licensed under the Zeebe Community License 1.1. You may not use this file
 * except in compliance with the Zeebe Community License 1.1.
 */
package io.camunda.zeebe.engine.state.immutable;

import io.camunda.zeebe.protocol.ZbColumnFamilies;
import io.camunda.zeebe.stream.api.StreamProcessorLifecycleAware;

public interface ProcessingState extends StreamProcessorLifecycleAware {

  DeploymentState getDeploymentState();

  ProcessState getProcessState();

  JobState getJobState();

  MessageState getMessageState();

  MessageSubscriptionState getMessageSubscriptionState();

  MessageStartEventSubscriptionState getMessageStartEventSubscriptionState();

  ProcessMessageSubscriptionState getProcessMessageSubscriptionState();

  IncidentState getIncidentState();

  BannedInstanceState getBannedInstanceState();

  VariableState getVariableState();

  TimerInstanceState getTimerState();

  ElementInstanceState getElementInstanceState();

  EventScopeInstanceState getEventScopeInstanceState();

  DecisionState getDecisionState();

  SignalSubscriptionState getSignalSubscriptionState();

  DistributionState getDistributionState();

  PendingMessageSubscriptionState getPendingMessageSubscriptionState();

  PendingProcessMessageSubscriptionState getPendingProcessMessageSubscriptionState();

  int getPartitionId();

  boolean isEmpty(final ZbColumnFamilies column);
}
