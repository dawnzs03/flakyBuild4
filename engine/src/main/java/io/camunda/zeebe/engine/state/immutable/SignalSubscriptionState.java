/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
 * one or more contributor license agreements. See the NOTICE file distributed
 * with this work for additional information regarding copyright ownership.
 * Licensed under the Zeebe Community License 1.1. You may not use this file
 * except in compliance with the Zeebe Community License 1.1.
 */
package io.camunda.zeebe.engine.state.immutable;

import io.camunda.zeebe.engine.state.signal.SignalSubscription;
import io.camunda.zeebe.protocol.impl.record.value.signal.SignalSubscriptionRecord;
import org.agrona.DirectBuffer;

public interface SignalSubscriptionState {

  boolean exists(SignalSubscriptionRecord subscription);

  void visitBySignalName(DirectBuffer signalName, SignalSubscriptionVisitor visitor);

  /**
   * Visit all subscriptions with the given process definition key.
   *
   * @param processDefinitionKey the key of the process definition the subscription belongs to
   * @param visitor the function that is called for each subscription
   */
  void visitStartEventSubscriptionsByProcessDefinitionKey(
      long processDefinitionKey, SignalSubscriptionVisitor visitor);

  @FunctionalInterface
  interface SignalSubscriptionVisitor {
    void visit(SignalSubscription subscription);
  }
}
