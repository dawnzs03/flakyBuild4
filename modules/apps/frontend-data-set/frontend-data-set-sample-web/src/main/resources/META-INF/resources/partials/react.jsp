<%--
/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */
--%>

<%@ include file="/init.jsp" %>

<%
ReactFDSDisplayContext reactFDSDisplayContext = new ReactFDSDisplayContext(request);
%>

<react:component
	module="{SampleReactFrontendDataSet} from frontend-data-set-sample-web"
	props='<%=
		HashMapBuilder.<String, Object>put(
			"apiURL", reactFDSDisplayContext.getAPIURL()
		).put(
			"id", FDSSampleFDSNames.REACT
		).put(
			"style", "fluid"
		).put(
			"views", reactFDSDisplayContext.getViews()
		).build()
	%>'
/>