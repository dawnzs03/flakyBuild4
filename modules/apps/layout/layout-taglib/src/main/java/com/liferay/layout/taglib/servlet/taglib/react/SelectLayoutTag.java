/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

package com.liferay.layout.taglib.servlet.taglib.react;

import com.liferay.layout.taglib.internal.servlet.ServletContextUtil;
import com.liferay.layout.taglib.internal.util.LayoutUtil;
import com.liferay.portal.kernel.json.JSONArray;
import com.liferay.portal.kernel.json.JSONFactoryUtil;
import com.liferay.portal.kernel.json.JSONUtil;
import com.liferay.portal.kernel.log.Log;
import com.liferay.portal.kernel.log.LogFactoryUtil;
import com.liferay.portal.kernel.model.Group;
import com.liferay.portal.kernel.model.LayoutConstants;
import com.liferay.portal.kernel.service.LayoutServiceUtil;
import com.liferay.portal.kernel.theme.ThemeDisplay;
import com.liferay.portal.kernel.util.GetterUtil;
import com.liferay.portal.kernel.util.HashMapBuilder;
import com.liferay.portal.kernel.util.ParamUtil;
import com.liferay.portal.kernel.util.WebKeys;
import com.liferay.portal.util.PropsValues;
import com.liferay.taglib.util.IncludeTag;

import java.util.Map;

import javax.servlet.http.HttpServletRequest;
import javax.servlet.jsp.PageContext;

/**
 * @author Eudaldo Alonso
 * @author Marko Cikos
 */
public class SelectLayoutTag extends IncludeTag {

	public String getItemSelectorReturnType() {
		return _itemSelectorReturnType;
	}

	public String getItemSelectorSaveEvent() {
		return _itemSelectorSaveEvent;
	}

	public String getNamespace() {
		return _namespace;
	}

	public boolean isCheckDisplayPage() {
		return _checkDisplayPage;
	}

	public boolean isEnableCurrentPage() {
		return _enableCurrentPage;
	}

	public boolean isMultiSelection() {
		return _multiSelection;
	}

	public boolean isPrivateLayout() {
		return _privateLayout;
	}

	public void setCheckDisplayPage(boolean checkDisplayPage) {
		_checkDisplayPage = checkDisplayPage;
	}

	public void setEnableCurrentPage(boolean enableCurrentPage) {
		_enableCurrentPage = enableCurrentPage;
	}

	public void setItemSelectorReturnType(String itemSelectorReturnType) {
		_itemSelectorReturnType = itemSelectorReturnType;
	}

	public void setItemSelectorSaveEvent(String itemSelectorSaveEvent) {
		_itemSelectorSaveEvent = itemSelectorSaveEvent;
	}

	public void setMultiSelection(boolean multiSelection) {
		_multiSelection = multiSelection;
	}

	public void setNamespace(String namespace) {
		_namespace = namespace;
	}

	@Override
	public void setPageContext(PageContext pageContext) {
		super.setPageContext(pageContext);

		setServletContext(ServletContextUtil.getServletContext());
	}

	public void setPrivateLayout(boolean privateLayout) {
		_privateLayout = privateLayout;
	}

	@Override
	protected void cleanUp() {
		super.cleanUp();

		_checkDisplayPage = false;
		_enableCurrentPage = false;
		_itemSelectorReturnType = null;
		_itemSelectorSaveEvent = null;
		_multiSelection = false;
		_namespace = null;
		_privateLayout = false;
	}

	@Override
	protected String getPage() {
		return _PAGE;
	}

	@Override
	protected void setAttributes(HttpServletRequest httpServletRequest) {
		try {
			httpServletRequest.setAttribute(
				"liferay-layout:select-layout:data", _getData());
		}
		catch (Exception exception) {
			_log.error(exception);
		}
	}

	private Map<String, Object> _getData() throws Exception {
		HttpServletRequest httpServletRequest = getRequest();

		ThemeDisplay themeDisplay =
			(ThemeDisplay)httpServletRequest.getAttribute(
				WebKeys.THEME_DISPLAY);

		String[] selectedLayoutIds = ParamUtil.getStringValues(
			httpServletRequest, "layoutUuid");

		return HashMapBuilder.<String, Object>put(
			"checkDisplayPage", _checkDisplayPage
		).put(
			"config",
			HashMapBuilder.<String, Object>put(
				"loadMoreItemsURL",
				themeDisplay.getPathMain() + "/portal/get_layouts"
			).put(
				"maxPageSize",
				GetterUtil.getInteger(
					PropsValues.LAYOUT_MANAGE_PAGES_INITIAL_CHILDREN)
			).build()
		).put(
			"groupId", themeDisplay.getScopeGroupId()
		).put(
			"itemSelectorReturnType", _itemSelectorReturnType
		).put(
			"itemSelectorSaveEvent", _itemSelectorSaveEvent
		).put(
			"multiSelection", _multiSelection
		).put(
			"namespace", _namespace
		).put(
			"nodes", _getLayoutsJSONArray(selectedLayoutIds, themeDisplay)
		).put(
			"privateLayout", _privateLayout
		).put(
			"selectedLayoutIds", selectedLayoutIds
		).build();
	}

	private JSONArray _getLayoutsJSONArray(
			String[] selectedLayoutIds, ThemeDisplay themeDisplay)
		throws Exception {

		Group group = themeDisplay.getScopeGroup();

		if ((_privateLayout && !group.hasPrivateLayouts()) ||
			(!_privateLayout && !group.hasPublicLayouts())) {

			return JSONFactoryUtil.createJSONArray();
		}

		return JSONUtil.put(
			JSONUtil.put(
				"children",
				LayoutUtil.getLayoutsJSONArray(
					_checkDisplayPage, _enableCurrentPage,
					themeDisplay.getScopeGroupId(), getRequest(),
					_itemSelectorReturnType, _privateLayout, 0,
					selectedLayoutIds, 0,
					GetterUtil.getInteger(
						PropsValues.LAYOUT_MANAGE_PAGES_INITIAL_CHILDREN))
			).put(
				"disabled", true
			).put(
				"expanded", true
			).put(
				"hasChildren", true
			).put(
				"icon", "home"
			).put(
				"id", "0"
			).put(
				"name", themeDisplay.getScopeGroupName()
			).put(
				"paginated",
				() -> {
					int layoutsCount = LayoutServiceUtil.getLayoutsCount(
						themeDisplay.getScopeGroupId(), _privateLayout,
						LayoutConstants.DEFAULT_PARENT_LAYOUT_ID);

					if (layoutsCount >
							PropsValues.LAYOUT_MANAGE_PAGES_INITIAL_CHILDREN) {

						return true;
					}

					return false;
				}
			));
	}

	private static final String _PAGE = "/select_layout/page.jsp";

	private static final Log _log = LogFactoryUtil.getLog(
		SelectLayoutTag.class);

	private boolean _checkDisplayPage;
	private boolean _enableCurrentPage;
	private String _itemSelectorReturnType;
	private String _itemSelectorSaveEvent;
	private boolean _multiSelection;
	private String _namespace;
	private boolean _privateLayout;

}