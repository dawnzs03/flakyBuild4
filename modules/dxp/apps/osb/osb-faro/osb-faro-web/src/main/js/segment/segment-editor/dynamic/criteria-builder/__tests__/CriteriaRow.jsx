import CriteriaRow from '../CriteriaRow';
import mockStore from 'test/mock-store';
import React from 'react';
import {cleanup, fireEvent, render} from '@testing-library/react';
import {DndProvider} from 'react-dnd';
import {fromJS, Map} from 'immutable';
import {HTML5Backend} from 'react-dnd-html5-backend';
import {Provider} from 'react-redux';

jest.unmock('react-dom');

describe('CriteriaRow', () => {
	afterEach(cleanup);

	it('should render', () => {
		const {container, getAllByText, getByText} = render(
			<Provider store={mockStore()}>
				<DndProvider backend={HTML5Backend}>
					<CriteriaRow
						criterion={{
							operatorName: 'sessions-filter',
							propertyName: 'context/referrer',
							rowId: 'row_0',
							touched: false,
							valid: true,
							value: fromJS({
								criterionGroup: {
									conjunctionName: 'and',
									criteriaGroupId: 'group_0',
									items: [
										{
											operatorName: 'ne',
											propertyName: 'context/referrer',
											rowId: 'row_1',
											touched: false,
											valid: true
										},
										{
											operatorName: 'gt',
											propertyName: 'completeDate',
											rowId: 'row_2',
											touched: false,
											valid: true
										}
									]
								}
							})
						}}
						referencedAssetsIMap={new Map()}
						referencedPropertiesIMap={new Map()}
					/>
				</DndProvider>
			</Provider>
		);

		fireEvent.click(getByText('is not'));

		expect(getByText('is')).toBeTruthy();
		expect(getAllByText('is not')[1]).toBeTruthy();
		expect(getByText('contains')).toBeTruthy();
		expect(getByText('does not contain')).toBeTruthy();
		expect(getByText('is known')).toBeTruthy();
		expect(getByText('is unknown')).toBeTruthy();
		expect(container).toMatchSnapshot();
	});

	it('should render w/ Non-Existent Property message', () => {
		const {queryByText} = render(
			<Provider store={mockStore()}>
				<DndProvider backend={HTML5Backend}>
					<CriteriaRow
						referencedAssetsIMap={new Map()}
						referencedPropertiesIMap={new Map()}
					/>
				</DndProvider>
			</Provider>
		);

		expect(queryByText('Attribute no longer exists.')).toBeTruthy();
	});
});
