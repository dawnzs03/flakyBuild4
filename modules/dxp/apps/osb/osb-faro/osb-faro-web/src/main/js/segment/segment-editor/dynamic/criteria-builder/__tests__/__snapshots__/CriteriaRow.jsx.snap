// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CriteriaRow should render 1`] = `
<div>
  <div
    class="criterion-row-root"
  >
    <div
      aria-hidden="true"
      class="color-stripe color--session"
      data-aria-hidden="true"
    />
    <div
      class="edit-container"
    >
      <div
        aria-hidden="true"
        class="drag-icon"
        data-aria-hidden="true"
        draggable="true"
      >
        <svg
          class="lexicon-icon lexicon-icon-drag icon-root"
          role="presentation"
        >
          <use
            xlink:href="#drag"
          />
        </svg>
      </div>
      <div
        class="criteria-statement"
      >
        <div
          class="form-group form-group-autofit"
        >
          <div
            class="criteria-statement"
          >
            <div
              class="form-group form-group-autofit"
            >
              <div
                aria-hidden="true"
                class="form-group-item entity-name form-group-item-label form-group-item-shrink"
                data-aria-hidden="true"
              >
                Session
              </div>
              <div
                aria-hidden="true"
                class="form-group-item display-value form-group-item-label form-group-item-shrink"
                data-aria-hidden="true"
              >
                Referrer
              </div>
              <div
                class="form-group-item form-group-item-shrink"
              >
                <button
                  aria-activedescendant="ne"
                  aria-expanded="true"
                  aria-haspopup="listbox"
                  aria-owns="clay-id-2"
                  class="form-control form-control-select form-control-select-secondary show"
                  role="combobox"
                  style=""
                  tabindex="0"
                  type="button"
                >
                  is not
                </button>
              </div>
              <div
                aria-hidden="true"
                class="form-group-item"
                data-aria-hidden="true"
              >
                <div
                  aria-haspopup="true"
                  class="input-group base-select-input-root select-input-root"
                >
                  <div
                    class="input-group-item"
                  >
                    <input
                      autocomplete="off"
                      class="input-root form-control input-group-inset input-group-inset-after"
                      placeholder=""
                      type="text"
                      value=""
                    />
                    <div
                      class="input-group-inset-item input-group-inset-item-after"
                    >
                      <svg
                        class="lexicon-icon lexicon-icon-caret-bottom icon-root"
                        role="presentation"
                      >
                        <use
                          xlink:href="#caret-bottom"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          aria-hidden="true"
          class="form-group form-group-autofit"
          data-aria-hidden="true"
        >
          <div
            class="form-group-item form-group-item-shrink"
          >
            <button
              aria-activedescendant=""
              aria-expanded="false"
              aria-haspopup="listbox"
              class="form-control form-control-select form-control-select-secondary conjunction-input"
              data-testid="conjunction-input"
              role="combobox"
              tabindex="0"
              type="button"
            >
              after
            </button>
          </div>
          <div
            class="form-group-item form-group-item-shrink"
          >
            <div
              aria-haspopup="true"
              class="input-group date-input-root"
            >
              <div
                class="input-group-item"
              >
                <input
                  autocomplete="off"
                  class="input-root form-control input-group-inset input-group-inset-after"
                  data-testid="date-input"
                  placeholder="YYYY-MM-DD"
                  readonly=""
                  type="text"
                  value=""
                />
                <div
                  class="input-group-inset-item input-group-inset-item-after"
                >
                  <button
                    aria-label="Choose a date"
                    class="button-root btn btn-unstyled"
                    type="button"
                  >
                    <svg
                      class="lexicon-icon lexicon-icon-calendar icon-root"
                      role="presentation"
                    >
                      <use
                        xlink:href="#calendar"
                      />
                    </svg>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        aria-hidden="true"
        class="actions"
        data-aria-hidden="true"
      >
        <div
          class="dropdown dropdown-action"
        >
          <button
            aria-controls="clay-dropdown-menu-1"
            aria-expanded="false"
            aria-haspopup="true"
            aria-label="Menu"
            class="dropdown-toggle component-action btn btn-unstyled"
            type="button"
          >
            <svg
              class="lexicon-icon lexicon-icon-ellipsis-v"
              role="presentation"
            >
              <use
                xlink:href="#ellipsis-v"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
